### ISSUE TITLE
psql \bind parameters should work

### ISSUE THREAD

**Issue by refset** [xtdb/crux#3572]
In principle I believe this should work in psql (and it would help with testing if it did):
```
jdt=> INSERT INTO tbl1 (_id, foo) VALUES ($1, $2) \bind 'a' 'b' \g
INSERT 0 0
```
But instead the logs give:
```
12:47:13 | DEBUG xtdb.indexer | aborted tx
xtdb.RuntimeException: 0 arguments were provided and 2 arguments were provided
        at xtdb.error$runtime_err.invokeStatic(error.clj:21)
        at xtdb.error$runtime_err.invoke(error.clj:13)
        at xtdb.error$runtime_err.invokeStatic(error.clj:17)
        at xtdb.error$runtime_err.invoke(error.clj:13)
        at xtdb.indexer$wrap_sql_args$fn__25961.invoke(indexer.clj:426)
        at xtdb.indexer$foreach_arg_row.invokeStatic(indexer.clj:413)
        at xtdb.indexer$foreach_arg_row.invokePrim(indexer.clj)
        at xtdb.indexer$$reify__25966.indexOp(indexer.clj:401)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019$fn__26032.invoke(indexer.clj:623)
        at clojure.lang.AFn.call(AFn.java:18)
        at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019.invoke(indexer.clj:620)
        at xtdb.indexer.Indexer$fn__26016$fn__26040.invoke(indexer.clj:634)
        at xtdb.indexer.Indexer$fn__26016.invoke(indexer.clj:633)
        at xtdb.indexer.Indexer.indexTx(indexer.clj:597)
        at xtdb.log.watcher$watch_log_BANG_$reify__26256.acceptRecord(watcher.clj:53)
        at xtdb.log$tx_handler$fn__17626.invoke(log.clj:37)
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
        at clojure.core.protocols$fn__8230.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8230.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at xtdb.log.NotifyingSubscriberHandler$fn__17659.invoke(log.clj:96)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.lang.Thread.run(Thread.java:1583)
```

Also:

1. Where it says "0 arguments were provided and 2 arguments were provided" I assume the first 'provided' there should actually be 'expected'
2. We should probably relabel 'arguments' as 'parameters' in the error messages given a general familiarity/expectation for the latter

**Comment by jarohen** [comment]
> Where it says "0 arguments were provided and 2 arguments were provided" I assume the first 'provided' there should actually be 'expected'

Heh, I ran into this separately - fixed in 16bfe1ac9875206f7b871d940d33a1071873be93

> We should probably relabel 'arguments' as 'parameters' in the error messages given a general familiarity/expectation for the latter

There's (/there should be) a [subtle distinction](https://en.wikipedia.org/wiki/Parameter_(computer_programming)) here - parameters are the static syntactic inputs, arguments are the dynamic values provided

**Comment by jarohen** [comment]
This now appears to be passing the arguments through - we're now getting the same error as #3622, so will close this in favour of the other card
